test/gui/shared/scripts/pageObjects/PublicLinkDialog.py

```diff
@@ -173,24 +229,22 @@ def createPublicLinkWithRole(self, role):
             )
         )
 
-    def changePassword(self, publicLinkName, password):
-        test.compare(
-            str(squish.waitForObjectExists(self.PUBLIC_LINK_NAME).text),
-            publicLinkName,
-        )
-        squish.mouseClick(
-            squish.waitForObject(names.oCC_ShareLinkWidget_lineEdit_password_QLineEdit),
-            0,
-            0,
-            squish.Qt.NoModifier,
-            squish.Qt.LeftButton,
-        )
-        squish.type(
-            squish.waitForObject(names.oCC_ShareLinkWidget_lineEdit_password_QLineEdit),
-            password,
-        )
+    def changePassword(self, password):
+        self.setPassword(password)
```

Comment:
I  think this function should remain the way it is as we check the link name(which might be useful when there are many public links for a file/folder) and edit the existing password but `setPassword` does not check the name of public link. 

Maybe you can call `verifyPublicLinkName` function here?

Reply:
the public link name is checked in this step `the user changes the password of public link...`
here: https://github.com/owncloud/client/blob/e4cabdb9bd1eb2323b2df3b8a923deb4c92aeac3/test/gui/shared/steps/steps.py#L1199